util/src/main/java/io/grpc/util/AdvancedTlsX509TrustManager.java

```diff
@@ -303,27 +317,32 @@ public static Builder newBuilder() {
     return new Builder();
   }
 
-  // The verification mode when authenticating the peer certificate.
+  /**
+   * The verification mode when authenticating the peer certificate.
+   */
   public enum Verification {
-    // This is the DEFAULT and RECOMMENDED mode for most applications.
-    // Setting this on the client side will do the certificate and hostname verification, while
-    // setting this on the server side will only do the certificate verification.
+    /**
+     * This is the DEFAULT and RECOMMENDED mode for most applications.
+     * Setting this on the client side performs both certificate and hostname verification, while
+     * setting it on the server side only performs certificate verification.
+     */
     CERTIFICATE_AND_HOST_NAME_VERIFICATION,
-    // This SHOULD be chosen only when you know what the implication this will bring, and have a
-    // basic understanding about TLS.
-    // It SHOULD be accompanied with proper additional peer identity checks set through
-    // {@code PeerVerifier}(nit: why this @code not working?). Failing to do so will leave
-    // applications to MITM attack.
-    // Also note that this will only take effect if the underlying SDK implementation invokes
-    // checkClientTrusted/checkServerTrusted with the {@code SSLEngine} parameter while doing
-    // verification.
-    // Setting this on either side will only do the certificate verification.
+    /**
+     * DANGEROUS: Use trusted credentials to verify the certificate, but clients will not verify the
+     * certificate is for the expected host. This setting is only appropriate when accompanied by
+     * proper additional peer identity checks set through SslSocketAndEnginePeerVerifier. Failing to
+     * do so will leave your applications vulnerable to MITM attacks.
+     * This setting has the same behavior on server-side as CERTIFICATE_AND_HOST_NAME_VERIFICATION.
+     */
     CERTIFICATE_ONLY_VERIFICATION,
-    // Setting is very DANGEROUS. Please try to avoid this in a real production environment, unless
-    // you are a super advanced user intended to re-implement the whole verification logic on your
-    // own. A secure verification might include:
-    // 1. proper verification on the peer certificate chain
-    // 2. proper checks on the identity of the peer certificate
+    /**
+     * This SHOULD be used by advanced user intended to implement the entire verification logic
```

Comment:
Why remove "DANGEROUS"? This is strictly more dangerous than CERTIFICATE_ONLY_VERIFICATION. In fact, it is now _encouraging_ using it ("oh, I'm an advanced user, so I SHOULD use this.")

Reply:
Agreed, I put it back